portality/scripts/article_cleanup_sync.py

```diff
@@ -0,0 +1,137 @@
+"""
+For each article in the DOAJ index:
+    * Checks that it has a corresponding journal, deletes it otherwise (if -d set)
+    * Ensures that the article in_doaj status is the same as the journal's
+    * Applies the journal's information to the article metadata as needed
+"""
+import esprit
+from portality import models
+from portality.util import unicode_dict
+from datetime import datetime
+from portality.core import app
+
+batch_size = 1000
+
+# Cache the Journals here, so we don't keep searching for them in the index
+journal_cache = {}
+
+
+def cleanup_articles(write_changes=False, prep_all=False):
+
+    # Connection to the ES index
+    conn = esprit.raw.make_connection(None, 'localhost', 9200, 'doaj')
```

Comment:
can you use the elasticsearch settings for host, port and index from app.config (portality/settings.py) instead? The live app server actually refers to "gateway:9200" when it talks to ES. The gateway machine does the round-robin of ES nodes this way.